From 59ba7ee48fd553de99fe6aae6e2b2e945eaa5a6a Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Thu, 4 Mar 2010 12:32:28 +0100
Subject: [PATCH] Do not overwrite $LANGUAGE and other locale variables

gdm does not have a facility to specify a list of languages for $LANGUAGE, so
do not break $LANGUAGE by forcing it to $LANG (they have a completely different
syntax, and different meanings). Also, if the system sets $LC_ALL (which
distros should never ever do), we should not clobber this.

$LINGUAS is not a runtime environment variable for locales, so just drop this.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=603853
Bug-Ubuntu: https://launchpad.net/bugs/407300
---
 data/Xsession.in |   28 ++--------------------------
 1 files changed, 2 insertions(+), 26 deletions(-)

diff --git a/data/Xsession.in b/data/Xsession.in
index 6c7761a..b2d98f2 100755
--- a/data/Xsession.in
+++ b/data/Xsession.in
@@ -146,35 +146,11 @@ fi
 
 unset XKB_IN_USE
 
-# Normalize languages, some places/distros screw us up in /etc/profile,
-# so in case the user did select a language
+# Overwrite $LANG from /etc/profile (and friends) with the one picked in the
+# greeter
 if [ -n "$GDM_LANG" ]; then
   LANG="$GDM_LANG"
   export LANG
-
-  if [ -n "$LC_ALL" ]; then
-    if [ "$LC_ALL" != "$LANG" ]; then
-      LC_ALL="$LANG"
-    fi
-  else
-    unset LC_ALL
-  fi
-
-  if [ -n "$LANGUAGE" ]; then
-    if [ "$LANGUAGE" != "$LANG" ]; then
-      LANGUAGE="$LANG"
-    fi
-  else
-    unset LANGUAGE
-  fi
-
-  if [ -n "$LINGUAS" ]; then
-    if [ "$LINGUAS" != "$LANG" ]; then
-      LINGUAS="$LANG"
-    fi
-  else
-    unset LINGUAS
-  fi
 fi
 
 # run all system xinitrc shell scripts.
-- 
1.7.0

